src/main/scala/optics/poly/AndThen.scala

```diff
@@ -13,11 +13,11 @@ import optics.internal.{Applicative, Id, Proxy, TraversalRes}
 
 object AndThen {
 
-  given as AndThen[
+  given AndThen[
     EPTraversal,
     EPTraversal,
     EPTraversal
-  ] {
+  ] with {
```

Comment:
I really hope to remove these in the end encoding for Monocle; they're really nasty and I think @julien-truffaut has a better idea.

Reply:
The main type inference problem with overloaded `andThen` is when we compose optics coming from a typeclass such as `at`, `index`, `each`, ... See an example here https://github.com/lampepfl/dotty/issues/8418

In Monocle 3, I am proposing to add dedicated methods on the optics classes, for example, `Optional` would have a method `at`, `index`, `each`, ... It is a bit annoying to maintain as it creates some duplication and it is not scalable. However, it does provide a nice user experience since users can use code completion to find those methods without magical imports.

Reply:
Yeah - I ran into that issue multiple times (it's works horribly even with `inline` and match types in dotty as well).